smartsim/_core/generation/generator.py

```diff
@@ -24,219 +24,224 @@
+        """Generates the root directory for a specific job instance.
 
-        Generate the file structure for a SmartSim experiment. This
-        includes the writing and configuring of input files for a
-        application.
+        :param job: The Job instance for which the root directory is generated.
+        :param job_index: The index of the Job instance (used for naming).
+        :returns: The path to the root directory for the Job instance.
+        """
+        job_type = f"{job.__class__.__name__.lower()}s"
+        job_path = self.root / f"{job_type}/{job.name}-{job_index}"
+        return pathlib.Path(job_path)
 
-        To have files or directories present in the created entity
-        directories, such as datasets or input files, call
-        ``entity.attach_generator_files`` prior to generation. See
-        ``entity.attach_generator_files`` for more information on
-        what types of files can be included.
+    def _generate_run_path(self, job: Job, job_index: int) -> pathlib.Path:
+        """Generates the path for the \"run\" directory within the root directory
+        of a specific Job instance.
 
-        Tagged application files are read, checked for input variables to
-        configure, and written. Input variables to configure are
-        specified with a tag within the input file itself.
-        The default tag is surronding an input value with semicolons.
-        e.g. ``THERMO=;90;``
+        :param job (Job): The Job instance for which the path is generated.
+        :param job_index (int): The index of the Job instance (used for naming).
+        :returns: The path to the \"run\" directory for the Job instance.
+        """
+        path = self._generate_job_root(job, job_index) / "run"
+        path.mkdir(exist_ok=False, parents=True)
+        return pathlib.Path(path)
 
+    def _generate_log_path(self, job: Job, job_index: int) -> pathlib.Path:
         """
-        generator_manifest = Manifest(*args)
+        Generates the path for the \"log\" directory within the root directory of a specific Job instance.
 
-        self._gen_exp_dir()
-        self._gen_feature_store_dir(generator_manifest.fss)
-        self._gen_entity_list_dir(generator_manifest.ensembles)
-        self._gen_entity_dirs(generator_manifest.applications)
+        :param job: The Job instance for which the path is generated.
+        :param job_index: The index of the Job instance (used for naming).
+        :returns: The path to the \"log\" directory for the Job instance.
+        """
+        path = self._generate_job_root(job, job_index) / "log"
+        path.mkdir(exist_ok=False, parents=True)
+        return pathlib.Path(path)
 
-    def set_tag(self, tag: str, regex: t.Optional[str] = None) -> None:
-        """Set the tag used for tagging input files
+    # make this protected
+    @staticmethod
+    def log_file(log_path: pathlib.Path) -> pathlib.Path:
```

Comment:
Looks like one missing TODO here:
```suggestion
    @staticmethod
    def _log_file(log_path: pathlib.Path) -> pathlib.Path:
```

Reply:
AH